tests/ztimer_xsec/main.c

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (C) 2021 TUBA Freiberg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup tests
+ * @{
+ *
+ * @file
+ * @brief    high level ztimer test application
+ *
+ * @author   Karl Fessel <karl.fessel@ovgu.de>
+ *
+ *
+ * @}
+ */
+
+#include <stdio.h>
+
+#include "ztimer.h"
+#include "mutex.h"
+
+#ifndef LOG_LEVEL
+#define LOG_LEVEL LOG_WARNING
+#endif
+
+#include "log.h"
+
+typedef struct named_lock {
+    char *name;
+    mutex_t mut;
+} named_lock_t;
+
+void release(void *arg);
+
+static named_lock_t sec_lock = { .name = "SEC", .mut = MUTEX_INIT_LOCKED };
+static named_lock_t msec_lock = { .name = "MSEC", .mut = MUTEX_INIT_LOCKED };
+static named_lock_t usec_lock = { .name = "USEC", .mut = MUTEX_INIT_LOCKED };
+
+static ztimer_t sec_tim = { .callback = release, .arg =  &sec_lock };
+static ztimer_t msec_tim = { .callback = release, .arg = &msec_lock };
+static ztimer_t usec_tim = { .callback = release, .arg = &usec_lock };
+void release(void *arg)
+{
+    named_lock_t *e = arg;
+
+    LOG_INFO("time: %" PRIu32 "\n", ztimer_now(ZTIMER_USEC));
```

Comment:
Beware: `LOG_INFO()` is implemented using `printf()`. The `print_str()` and `print_u32_dec()` functions provided by `fmt` are quite stack efficient. If you really want to keep the printing from ISR. Alternatively you could just store the unlock time into a variable and use `printf()` from thread context, accessing this variable.

Reply:
since the LOG_INFOs are just for human driven testing on demand, i think they are able extend the ISR stacksize

Reply:
the loglevel is changed that infos are not printed

Reply:
I tend to agree with @maribu. It would actually be better to save the timestamp, e.g. extend your named mutex strucutre:

```
typedef struct named_lock {
    char *name;
    mutex_t mut;
    uint32_t time;
} named_lock_t;
```

Save the timestamp from in the ISR and then:

```
mutex_lock(&sec_lock.mut);
LOG_INFO("time: %" PRIu32 "\n", sec_lock.mut.time);
```

Logging in ISR can be useful for debugging, but should be avoided if there is another way around it and here there clearly is.

Reply:
Also the default logging level is `LOG_INFO` so why not use `LOG_DEBUG` for prints you do not want to show up by default?

Reply:
one may just want to add CFLAGS += -DLOG_LEVEL=LOG_INFO to the Makefile or make command line and get the information (without getting debug information of other modules)

Reply:
> one may just want to add CFLAGS += -DLOG_LEVEL=LOG_INFO to the Makefile or make a command line and get the information (without getting debug information of other modules)

Why not leave the default log level? Its already `LOG_INFO` https://github.com/RIOT-OS/RIOT/blob/fa9a297e7cecdf6761dc904b4ad30a8d30e88fb5/core/include/log.h#L66-L71

The is no harm in having this information logged, the automatic test will still pass (`expect_exact` will match the desired output it ignores other stdio output), for this use case ( a test ) conditionally setting unsettling logs overcomplicated things, I honestly don't think its worth it, I would as I said before just use `printf`. But I guess that is your choice as a contributor. But please double check that that CFLAG is necessary, I'm under the impression it isn't.

Reply:
I did not know that `expect_exact` just ignores extra output (I saw most tests being very sparse in the information they provide and thought it was to keep the expect script small), i thought all `expect_exact` have to match and something else being written to `stdout` will make the test fail.

I changed the LOG to printf, just provide the information without automatic checking.